CRM/ManualDirectDebit/Upgrader.php

```diff
@@ -7,11 +8,124 @@
 class CRM_ManualDirectDebit_Upgrader extends CRM_ManualDirectDebit_Upgrader_Base {
 
   public function install() {
+
     $this->executeSqlFile('sql/install.sql');
+    $this->createDirectDebitPaymentProcessorType();
+    $this->createDirectDebitPaymentProcessor();
+    $result = civicrm_api3('ContributionPage', 'create', [
+      'financial_type_id' => 1,
+      'is_recur' => 1,
+      'payment_processor' => "Direct Debit",
+      'title' => "Renew offline auto-renewal memberships",
+    ]);
   }
 
   public function uninstall() {
     $this->executeSqlFile('sql/uninstall.sql');
+    $this->uninstallCustomInformation();
+  }
+
+  /**
+   *  Uninstall custom information
+   */
+  private function uninstallCustomInformation() {
+    $customValuesForUninstall = [
+      [
+        "entityType" => "OptionGroup",
+        "searchValue" => "direct_debit_codes",
+      ],
+      [
+        "entityType" => "OptionGroup",
+        "searchValue" => "direct_debit_originator_number",
+      ],
+      [
+        "entityType" => "CustomGroup",
+        "searchValue" => "direct_debit_mandate",
+      ],
+      [
+        "entityType" => "CustomGroup",
+        "searchValue" => "direct_debit_information",
+      ],
+      [
+        "entityType" => "UFGroup",
+        "searchValue" => "Direct Debit Information",
+        "searchField" => "title",
+      ],
+      [
+        "entityType" => "PaymentProcessor",
+        "searchValue" => "OfflineDirectDebit",
+        "searchField" => "payment_processor_type_id",
+      ],
+      [
+        "entityType" => "PaymentProcessorType",
+        "searchValue" => "OfflineDirectDebit",
+      ],
+      [
+        "entityType" => "ContributionPage",
+        "searchValue" => "Renew offline auto-renewal memberships",
+        "searchField" => "title",
+      ],
+    ];
+
+    foreach ($customValuesForUninstall as $customValue) {
+      $this->deleteEntityRecord(
+        $customValue['entityType'],
+        $customValue['searchValue'],
+        isset($customValue['searchField']) ? $customValue['searchField'] : ""
+      );
+    }
+  }
+
+  /**
+   * Deletes a record for the specified CiviCRM entity based on
+   * the search field and search value.
+   *
+   * @param string $entityType
+   *   The entity type for the record we want to remove (e.g OptionValue,
+   *   PaymentProcessor ..etc). it should be a valid Custom or CiviCRM core
+   *   entity type.
+   * @param string $searchValue
+   *   The search value to find the record that we want to remove
+   * @param string $searchField
+   *   The field that we search the search value against.
+   */
+  private function deleteEntityRecord($entityType, $searchValue, $searchField = 'name') {
+    civicrm_api3($entityType, 'get', [
+      $searchField => $searchValue,
+      'api.' . $entityType . '.delete' => ['id' => '$value.id'],
+    ]);
+  }
+
+  /**
+   * Will install the DirectDebit payment processor
```

Comment:
Please rename the method description to : 

Installs the 'Direct Debit' payment processor Type